lily4178993/maths-magicians
# Add Unit Tests for React Components and Calculator Logic Functions

# Add Unit Tests for React Components and Calculator Logic Functions

## Description
This pull request includes comprehensive unit tests for both React components and the calculator logic functions used in the project. The goal is to ensure the correctness, functionality, and reliability of these components and logic.

### React Components Tested
- `Menu` component: Tests for rendering navigation links.
- `CalcOutput` component: Tests for enforcing PropTypes.
- `CalcButton` component: Tests for rendering buttons and handling click events.
- `CalcKeyboard` component: Tests for rendering the keyboard layout.
- `Quotes` component: Tests for rendering quotes, loading messages.
- `CalculatorPage`: Tests for rendering and functionality.
- `Home`: Tests for rendering and content.
- `NotFound`: Tests for rendering and content.
- `Quote`: Tests for rendering and asynchronous behavior.

### Calculator Logic Functions Tested
- `calculate.js`: Unit tests for various scenarios.
- `operate.js`: Unit tests for all operations.

## Checklist
- [x] Added unit tests for multiple React components and calculator logic.
- [x] Utilized Jest snapshots for component rendering.
- [x] Simulated user interactions using React Testing Library.
- [x] Ensured all tests pass successfully.
- [x] Maintained or improved code coverage.
- [x] Ensured code is well-documented and adheres to coding standards.


## How to Test

To thoroughly test this branch, follow these steps:

1. Ensure you have cloned this branch and installed dependencies.
2. Ensure you are in the root directory of the project.
3. Execute the following command to run React component and calculator logic function  tests:
   ```bash
   npm test a
   ```
4. Review the test results in your terminal. Ensure that all React component tests pass successfully. These tests cover components such as `Menu`, `CalcOutput`, `CalcButton`, `CalcKeyboard`, `Quotes`, `CalculatorPage`, `Home`, `NotFound`, and `Quote`.
5. Confirm that all calculator logic function tests pass successfully. These tests validate the correctness of the `calculate.js `and `operate.js` files.
